src/neo/SmartContract/ApplicationEngine.cs

```diff
@@ -302,19 +302,31 @@ protected void ValidateCallFlags(CallFlags requiredCallFlags)
                 throw new InvalidOperationException($"Cannot call this SYSCALL with the flag {state.CallFlags}.");
         }
 
-        protected override void OnSysCall(uint method)
+        protected void OnSysCall(InteropDescriptor descriptor, Func<object, object[], object> handler)
```

Comment:
Why `Func<object, object[], object>` instead of `descriptor.Handler`?

Reply:
So the debugger can override the behavior of existing interop services such as `System.Runtime.CheckWitness`

Example: https://github.com/ngdenterprise/neo-blockchaintoolkit-library/blob/b16214cfc059a4175311dbbb3bd67a90b4ecd144/src/bctklib-3/smart-contract/TestApplicationEngine.cs#L80

Reply:
Please check #2382. I think it can solve this problem.

Reply:
I can make #2382 work